app/models/journeyDomain/transportMeans/TransportMeansActiveDomain.scala

```diff
@@ -35,19 +36,18 @@ import play.api.i18n.Messages
 import play.api.mvc.Call
 
 sealed trait TransportMeansActiveDomain extends JourneyDomainModel {
-
-  val identification: Identification
-  val identificationNumber: String
-
-  def asString(implicit messages: Messages): String =
-    TransportMeansActiveDomain.asString(identification, identificationNumber)
-
+  def asString(implicit messages: Messages): String
 }
 
 object TransportMeansActiveDomain {
 
-  def asString(identification: Identification, identificationNumber: String)(implicit messages: Messages): String =
-    s"${identification.asString} - $identificationNumber"
+  def asString(identification: Option[Identification], identificationNumber: Option[String])(implicit messages: Messages): String =
+    (identification, identificationNumber) match {
+      case (Some(identification), Some(identificationNumber)) => s"${identification.asString} - $identificationNumber"
+      case (Some(identification), None)                       => identification.asString
+      case (None, Some(identificationNumber))                 => identificationNumber
+      case (None, None)                                       => "" // TODO - what should this be?
```

Comment:
Should this raise iff mandatory or can it be nil if it can't be worked out?

Reply:
Working on a refactor for this now. This eventuality would only arise during transition where both questions are optionals, but we don't have multiplicity during transition so this method would never actually be called